### PR TITLE
Support ids

### PR DESCRIPTION
**Pull request by aaronranard**
Change name to be id and v`2.0.0` will always require an id

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -52,55 +53,56 @@ export default function compile(services: Services, blocks: Blocks): GraphResult
 
   Object.keys(services).forEach((key) => {
     const {
+      id,
       name,
```
Comment (tlugger) [inline src/index.js]: Looks like we can remove `name` here since we never use it afterwords.

File: src/index.js [hunk 2]
```diff
@@ -52,55 +53,56 @@ export default function compile(services: Services, blocks: Blocks): GraphResult
...
       execution,
       mapping,
     }: ServiceShape = services[key];
 
-    nodes.add(name);
+    nodes.add(id);
 
     const mappings: Mapping = mapping ?
-      mapping.reduce((rest, { name: alias, mapping: block }) => ({
+      mapping.reduce((rest, { id, mapping: block }) => ({
```
Comment (tlugger) [inline src/index.js]: I'm getting `no-shadow` linter errors for this line.

File: src/index.js [hunk 3]
```diff
@@ -52,55 +53,56 @@ export default function compile(services: Services, blocks: Blocks): GraphResult
...
         ...rest,
-        [alias]: block,
+        [id]: block,
       }), {}) : {};
 
-    const resolveName = alias => (mappings[alias] || alias);
+    const resolveId = id => (mappings[id] || id);
```
Comment (tlugger) [inline src/index.js]: Same for this line. Not sure why because we never change the value of `id` it looks like...

**Review by tlugger — CHANGES_REQUESTED** [review]

**Comment by aaronranard** [comment]
Addressed linter issues

**Review by tlugger — APPROVED** [review]

**Pull request merged**